### PR TITLE
Add support for openapi-contract-first development

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

**Comment by jamesnetherton** [comment]
For the templates used to generate the model sources, how likely is it that we'll need to update them frequently?

I guess it'd just be a case of copying them from the original Maven plugin source?

File: extensions/rest-openapi/deployment/src/main/java/org/apache/camel/quarkus/component/rest/openapi/deployment/CamelQuarkusSwaggerCodegenProvider.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                            .filter(Files::isRegularFile)
+                            .filter(s -> s.toString().endsWith("json"))
+                            .map(Path::normalize)
+                            .map(Path::toAbsolutePath)
+                            .map(Path::toString)
+                            .forEach(jsonFiles::add);
+                }
+            }
+
+            String packageName = config.getValue("camel.rest.bindingPackageScan", String.class);
```
[the remark this answers is not in the document]
Comment (zhfeng) [reply extensions/rest-openapi/deployment/src/main/java/org/apache/camel/quarkus/component/rest/openapi/deployment/CamelQuarkusSwaggerCodegenProvider.java:78]: Good catch ! and I agree to add such a config option for model package.

File: extensions/rest-openapi/runtime/src/main/java/org/apache/camel/quarkus/rest/openapi/runtime/RestOpenApiBuildTimeConfig.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "camel.openapi", phase = ConfigPhase.BUILD_TIME)
+public class RestOpenApiBuildTimeConfig {
+    /**
+     * Build time configuration options for Camel Quarkus gRPC code generator.
```
[the remark this answers is not in the document]
Comment (zhfeng) [reply extensions/rest-openapi/runtime/src/main/java/org/apache/camel/quarkus/rest/openapi/runtime/RestOpenApiBuildTimeConfig.java]: Fixed

File: extensions/rest-openapi/runtime/src/main/java/org/apache/camel/quarkus/rest/openapi/runtime/RestOpenApiBuildTimeConfig.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    @ConfigItem
+    public CodeGenConfig codegen;
+
+    @ConfigGroup
+    public static class CodeGenConfig {
+        /**
+         * If {@code true}, Camel Quarkus OpenApi code generation is run for .json files discovered from the {@code openapi}
+         * directory. When * {@code false}, code generation for .json files is disabled.
```
[the remark this answers is not in the document]
Comment (zhfeng) [reply extensions/rest-openapi/runtime/src/main/java/org/apache/camel/quarkus/rest/openapi/runtime/RestOpenApiBuildTimeConfig.java]: Fixed

File: extensions/rest-openapi/deployment/src/main/java/org/apache/camel/quarkus/component/rest/openapi/deployment/CamelQuarkusSwaggerCodegenProvider.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhfeng) [reply extensions/rest-openapi/deployment/src/main/java/org/apache/camel/quarkus/component/rest/openapi/deployment/CamelQuarkusSwaggerCodegenProvider.java]: Added

**Comment by zhfeng** [comment]
I think it could be not changed very offen. I only copy some templates which are related to the models generation. They are copied from `io.swagger.codegen.v3:swagger-codegen-generators`. And the only change is to add `@RegisterForReflection` in the `generatedAnnotation.mustache` with
```java
@RegisterForReflection{{#serializableModel}}(serialization = true){{/serializableModel}}
```

**Pull request merged**